subsys/usb/device_next/class/usbd_uvc.c

```diff
@@ -1674,22 +1664,13 @@ static int uvc_init(struct usbd_class_data *const c_data)
 				uvc_assign_desc(dev, &cfg->desc->if1_color, true, true);
 			}
 
-			ret = uvc_add_vs_format_desc(dev, &format_desc, cap);
-			if (ret != 0) {
-				return ret;
-			}
+			uvc_add_vs_format_desc(dev, &format_desc, cap);
```

Comment:
I am not sure it makes sense. If due to wrong configuration the UVC function will be non functional, then it should fail during initialization. Otherwise I think it is okay but I would keep the return values above, and then log them and not fail here.

Reply:
But I am still curious, will the UVC be functional if CONFIG_USBD_VIDEO_MAX_FORMATS or CONFIG_USBD_VIDEO_MAX_FRMIVAL are set too low?

Reply:
It would present fewer formats to the host, and the host would still be able to select some.

It could eventually lead to one invalid format descriptor with no frame/resolution descriptor for it, or it could lead to a list of descriptors that lack a few supported format but still valid as per USB standard.

I could as well return the error directly if this makes more sense to ensure "descriptors guaranteed correct, or nothing at all".

Guidance appreciated!